fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java

```
@@ -1238,6 +1285,14 @@ private void handleSuggestedDomains(SuggestDomainsResponsePayload payload) {
         emitChange(event);
     }
 
+    private void fetchPlans(SiteModel siteModel) {
+        mSiteRestClient.fetchPlans(siteModel);
```

The passed site could be a self-hosted site in which case the rest client will return an error. We need to handle that with a special error type. [Here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/store/ThemeStore.java#L322-L330) is an example.

Unfortunately we have not been very consistent with this and in some cases we just swallow the error and there is no callback event. However, we have been trying to do a better job with that.

---

Thanks. I hadn't known/realized this scenario at all.
I have introduced a new error type `PlansErrorType.NOT_AVAILABLE` and handled it the same way as in the example.